corehq/apps/users/models.py

```diff
@@ -652,6 +652,9 @@ def has_permission(self, domain, permission, data=None, restrict_global_admin=Fa
 
         dm = self.get_domain_membership(domain)
         if dm:
+            # an admin has access to all features by default, restrict that if needed
+            if dm.is_admin and restrict_global_admin:
+                return False
```

Comment:
I think you need to pass `restrict_global_admin` along to `DomainMembership.has_permission`. This code will return false even if the user does specifically have the permssions assigned to them.

Reply:
That's what I'm saying: pass `restrict_global_admin` into that function and change it from 
`return self.is_admin or self.permissions.has(permission, data)`
to
`return (self.is_admin and not restrict_global_admin) or self.permissions.has(permission, data)`

Right now if `restrict_global_admin` is true, there's no way for *this* `has_permission` to return true, which seems wrong.

Reply:
I was wrong about this, nevermind.